### DISCUSSION TITLE
M52 stuck in DFU mode

### DISCUSSION THREAD

**Discussion by ZsomBEAST** [tinymovr/Tinymovr#370] [Q&A - CANine Adapter]
Hello Guys! I am new with Tinymovr and I am stucked with an issue. I have just bought 4 pcs Tinymovr M52. I have cloned the sourcecode of the firmware (2.3.1) from your git. I am tring to use my own HALL sensor through SPI, so I implemented my driver, then built the new firmware. I am using a CANine V14 to flash my M52 through CAN. I can flash the device successfully, but after that I am not able to exit the DFU mode (tinmymovr app detect the device as tm_dfu1). I have tried to reset and repower it, but nothing worked. I have also tried to flash your newest 2.3.1 firmware, but still can't exit DFU mode. I have tried with another board, and happens the same thing. Can you help me solve this issue?

**Comment by yconst** [comment]
Please make sure you are building/uploading the "update"  target, instead of the "release" target, this is what is compatible with DFU

**Reply by ZsomBEAST** [reply]
Yes, I have used the "tinymovr-5fadde7-dirty-M51-upgrade.bin" from the binaries I have built. Maybe my costume firmware was bugged, but since that I can't exit the DFU mode so I can't really debug it. Since that I have tried the original "tinymovr-2.3.1-M51-upgrade.bin" as well. I can flash it to the device, but still can't exit the DFU mode.

**Reply by yconst** [reply]
Try a full erase from the DFU menu. This should clear all flash except the bootloader region, then try re-uploading the stock firmware using DFU
